testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/RemoteEJBClientTwoClustersTestCase.java

```diff
@@ -0,0 +1,380 @@
+        return (NODE_3.equals(nodeName) || NODE_4.equals(nodeName));
+    }
+
+    private boolean hasWeakAffinityNONE(Incrementor proxy) {
+        return EJBClient.getWeakAffinity(proxy).equals(Affinity.NONE);
+    }
+
+    private void logConfiguredConnections() {
+        logger.info("Checking configured connections:");
+        for (EJBClientConnection connection : EJBClientContext.getCurrent().getConfiguredConnections()) {
+            logger.info("connection: " + connection.getDestination());
+        }
+    }
+
+    /**
+     * Set up the JNDI environment to allow SFSB creation on the correct cluster
+     *
+     * @param clusterName
+     * @return
+     */
+    private Properties getJNDIEnvironmentForCluster(String clusterName) {
+        Properties env = new Properties();
+        env.setProperty(Context.INITIAL_CONTEXT_FACTORY, org.wildfly.naming.client.WildFlyInitialContextFactory.class.getName());
+        env.setProperty("jboss.cluster-affinity", clusterName);
+        return env;
+    }
+
+    /**
+     * Setup server configurations to allow two independent clusters: clusterA = {node1, node2}, clusterB = {node3, node4}
+     * This requires the following changes:
+     * 1. keeping the two clusters separated at the JGroups level:
+     *   - the server's JChannel managed by resource channel "ee" needs to have a different name in each cluster
+     *   - the discovery protocol used by JGroups needs to use a different discovery address in each cluster
+     * 2. providing distinct cluster identities used for ClusterAffinity processing which are provided by topology updates
+     *   - for distributed caches over a single channel, the required ClusterAffinity value defaults to the JChannel name of the channel used
+     */
+    public static class ServerSetupTask extends CLIServerSetupTask {
+        public ServerSetupTask() {
+            this.builder
+                    // clusterA nodes
+                    .node(NODE_1, NODE_2)
+                    // distinct JChannel name for each node in cluster A
+                    .setup("/subsystem=jgroups/channel=ee:write-attribute(name=cluster,value=clusterA)")
+                    .teardown("/subsystem=jgroups/channel=ee:write-attribute(name=cluster,value=ejb)")
+                    .parent()
+                    // clusterB nodes
+                    .node(NODE_3, NODE_4)
+                    // distinct JChannel name in cluster B for each node as well as unique discovery address
+                    .setup("/subsystem=jgroups/channel=ee:write-attribute(name=cluster,value=clusterB)")
+                    .setup("/socket-binding-group=standard-sockets/socket-binding=jgroups-mping:write-attribute(name=multicast-address,value=%s)", TESTSUITE_MCAST3)
+                    .teardown("/socket-binding-group=standard-sockets/socket-binding=jgroups-mping:write-attribute(name=multicast-address,value=%s", TESTSUITE_MCAST)
```

Comment:
There's a missing closing bracket in the `:write-attribute(..)` operation, but the client tolerates this.

Reply:
Done.